test/internal/register_internal_actor.js

```diff
@@ -0,0 +1,140 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const utils = require('../test_lib/utils'),
+  { Event } = require('../test_lib/event_decoder'),
+  Internal = artifacts.require('Internal'),
+  InternalUtils = require('./utils'),
+  AccountProvider = utils.AccountProvider;
+
+contract('Internal::registerinternalactor', async (accounts) => {
+
+  let internal,
+    organization,
+    accountProvider,
+    worker;
+
+  beforeEach(async function() {
+
+    accountProvider = new AccountProvider(accounts);
+    ({ internal, worker, organization } = await InternalUtils.setupInternal(accountProvider));
+
+  });
+
+  describe('Negative Tests', async () => {
+
+    it('Reverts if non-worker address is adding internal actor', async () => {
+
+      let internalActors = [];
+        internalActors.push(accountProvider.get());
+        let nonWorker = accountProvider.get();
+
+      await utils.expectRevert(internal.registerInternalActor(
+        internalActors,
+        { from: nonWorker },
+        ),
+        'Worker should be registered.',
+        'Only whitelisted worker is allowed to call.'
+      );
+
+    });
+
+  });
+
+  describe('Events', async (accounts) => {
+
+    it('Verify InternalActorRegistered event', async () => {
+
+      let internalActors = [];
+      internalActors.push(accountProvider.get());
+      internalActors.push(accountProvider.get());
+      internalActors.push(accountProvider.get());
+
+      let transactionResponse = await internal.registerInternalActor(
+        internalActors,
+        { from: worker },
+      );
+
+      const events = Event.decodeTransactionResponse(
+        transactionResponse,
+      );
+
+      Event.assertEqualMulti(events,[ { name: 'InternalActorRegistered',
+          args:
+            { _organization: organization,
+              _actor: internalActors[0] } },
+          { name: 'InternalActorRegistered',
+            args:
+              { _organization: organization,
+                _actor: internalActors[1] } },
+          { name: 'InternalActorRegistered',
+            args:
+              { _organization: organization,
+                _actor: internalActors[2]} }
+        ],
+      );
+
+    });
+
+    it('Do not register already registered internal actor', async () => {
+
+      let internalActors = [];
+      internalActors.push(accountProvider.get());
+
+      await internal.registerInternalActor(
+        internalActors,
+        { from: worker },
+      );
+
+      internalActors.push(accountProvider.get());
+      let transactionResponse = await internal.registerInternalActor(
+        internalActors,
+        { from: worker },
+      );
+
+      const events = Event.decodeTransactionResponse(
+        transactionResponse,
+      );
+
+      Event.assertEqualMulti(events,[ { name: 'InternalActorRegistered',
```

Comment:
How the test case is verifying that already registered actor is not being added again?
1 way to test is no event is emitted when no registration is happening. Howcome the event is being emitted here?

Reply:
It was adding an another internal actor and was verifying that previously added internal actors are not emitted in InternalActorRegistered event.